File: FitnessNation/app/src/main/java/com/example/fitnessnation/activitites/LoginActivity.java
package com.example.fitnessnation.activitites;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.example.fitnessnation.AppDatabase;
import com.example.fitnessnation.ApplicationController;
import com.example.fitnessnation.FitnessRepository;
import com.example.fitnessnation.Meal;
import com.example.fitnessnation.R;
import com.example.fitnessnation.fragments.MenuLoginFragment;

public class LoginActivity extends AppCompatActivity {


    public static FitnessRepository fitnessRepository;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        fitnessRepository=new FitnessRepository(getApplicationContext());
        //fitnessRepository.mealInsertTask(Meal.populateData());



        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction().replace(R.id.primary_frame_s_session, MenuLoginFragment.newInstance()).commitNow();
        }



    }





}

File: FitnessNation/app/src/main/java/com/example/fitnessnation/activitites/LoginActivity.kt
package com.example.fitnessnation.activitites

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.room.Room
import com.example.fitnessnation.AppDatabase
import com.example.fitnessnation.LogInFragment
import com.example.fitnessnation.R
import com.example.fitnessnation.fragments.MenuLoginFragment

class LoginActivity : AppCompatActivity() {

   private var appDatabase: AppDatabase?=null
       get() = field


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        setFirstFragment()
       appDatabase = Room.databaseBuilder<AppDatabase>(
            applicationContext,
            AppDatabase::class.java, "userdatabase"
        ).allowMainThreadQueries().build()
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.container, LogInFragment.newInstance()).commitNow()
        }

    }

    fun setFirstFragment() {
        val fragment: MenuLoginFragment = MenuLoginFragment.newInstance()

        val transaction =supportFragmentManager.beginTransaction()
      transaction.add(R.id.primary_frame_s_session, fragment)
        transaction.addToBackStack(MenuLoginFragment.toString())
        transaction.commit()
    }


}
